velox/exec/TraceUtil.h

```diff
@@ -142,4 +141,39 @@ folly::dynamic getTaskMetadata(
 
 /// Checks whether the operator can be traced.
 bool canTrace(const std::string& operatorType);
+
+class DummySourceNode : public core::PlanNode {
+ public:
+  explicit DummySourceNode(const RowTypePtr outputType)
+      : PlanNode(""), outputType_(std::move(outputType)) {}
+
+  const RowTypePtr& outputType() const override {
+    return outputType_;
+  }
+
+  const std::vector<core::PlanNodePtr>& sources() const override;
+
+  std::string_view name() const override;
+
+  folly::dynamic serialize() const override;
+
+  static core::PlanNodePtr create(const folly::dynamic& obj, void* context) {
+    return std::make_shared<DummySourceNode>(
+        ISerializable::deserialize<RowType>(obj["outputType"]));
+  }
+
+ private:
+  void addDetails(std::stringstream& stream) const override {
+    // Nothing to add.
+  }
+
+  const RowTypePtr outputType_;
+};
+
+void registerDummySourceSerDe();
+
+/// Copy the target plan node, and replay the source node with DummySourceNode.
+core::PlanNodePtr copyTraceNode(
```

Comment:
```
/// Gets the plan node for trace with given plan node id. It replaces the actual sources with the dummy sources to avoid tracing unnecessary part of the plan.
core::PlanNodePtr getPlanNodeForTrace(
    const core::PlanNodePtr& rootNode,
    core::PlanNodeId traceNodeId);
```